base_agent/task.py

```diff
@@ -28,56 +25,154 @@ class Task(object):
         >>> Task()
     """
 
-    def __init__(self):
-        self.memid = None
+    def __init__(self, agent):
+        self.agent = agent
         self.interrupted = False
         self.finished = False
         self.name = None
         self.undone = False
         self.last_stepped_time = None
-        self.throttling_tick = DEFAULT_THROTTLING_TICK
+        self.prio = -1
+        self.running = 0
+        self.memid = TaskNode.create(self.agent.memory, self)
+        # TODO put these in memory in a new table?
+        # TODO methods for safely changing these
         self.stop_condition = NeverCondition(None)
+        self.on_condition = AlwaysCondition(None)
+        self.remove_condition = TaskStatusCondition(agent, self.memid)
+        self.child_generator = TaskGenerator()
+        TaskNode(agent.memory, self.memid).update_task(task=self)
 
-    def step(self, agent):
-        """The actual execution of a single step of the task is defined here."""
-        # todo? make it so something stopped by condition can be resumed?
-        if self.stop_condition.check():
-            self.finished = True
-            return
-        return
+    def check_remove_and_running_children(stepfn):
```

Comment:
docstring here as well. Why is this just a shell for `modified_step`?

Reply:
this a wrapper so we don't have to write the check for children etc every time we write a new task step; it takes in a stepfn (function) and returns a modified_step (function)